go/porcelain/deploy.go

```diff
@@ -383,16 +385,19 @@ func (n *Netlify) uploadFiles(ctx context.Context, d *models.Deploy, files *depl
 	for _, sha := range required {
 		if files, exist := files.Hashed[sha]; exist {
 			for _, file := range files {
-				sem <- 1
-				wg.Add(1)
-
-				go n.uploadFile(ctx, d, file, observer, t, timeout, wg, sem, sharedErr)
+				select {
+				case sem <- 1:
+					sem <- 1
+					wg.Add(1)
+
+					go n.uploadFile(ctx, d, file, observer, t, timeout, wg, sem, sharedErr)
+				case <-ctx.Done():
```

Comment:
memo: context cancellation can happen from the buildbot side (e.g. with timeout)